{{ cookiecutter.formal_name }}/{{ cookiecutter.formal_name }}/main.m

```diff
@@ -208,3 +186,61 @@ void crash_dialog(NSString *details) {
     // Show the crash dialog
     [alert runModal];
 }
+
+
+NSString * format_traceback(PyObject *type, PyObject *value, PyObject *traceback) {
+    
+    NSRegularExpression *regex;
+    NSString *traceback_str;
+    PyObject *traceback_list;
+    PyObject *traceback_module;
+    PyObject *format_exception;
+    PyObject *traceback_unicode;
+    
+
+    if (traceback == NULL) {
+        NSLog(@"Could not retrieve traceback");
+        return NULL;
+    }
+
+    // Drop the top two stack frames; these are internal
+    // wrapper logic, and not in the control of the user.
+    for (int i = 0; i < 2; i++) {
+        if (PyObject_GetAttrString(traceback, "tb_next") != NULL) {
```

Comment:
This is unlikely to fail - but if it does, it will break the call to `format_exception()`, so we should catch the else case and report the problem.

Reply:
Not sure what you mean here. How will `PyObject_GetAttrString` fail other than returning NULL? I guess there is a race condition here where the if-clause returns a PyObject and the following block returns NULL.

Reply:
If does fail, we are just left with the original stack frames.

Reply:
Ah - That's a good point. I was anticipating that the absence of `tb_next` would indicate a problem; but you're correct  - this code is effectively "retrieve 2 levels of tb_next if they're available; if they're not, just use the top-most available traceback object".